src/targets/gpu/fuse_mlir.cpp

```diff
@@ -301,14 +301,87 @@ struct find_mlir_op
     }
 };
 
+struct find_mlir_standalone_convolution_op
+{
+    auto matcher() const { return match::name("convolution"); }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        auto conv_based_op = r.result;
+        // enable only for fp32/fp16/i8 types
+        if(std::any_of(conv_based_op->inputs().begin(), conv_based_op->inputs().end(), [&](auto i) {
+               return not contains(
+                   {shape::type_t::float_type, shape::type_t::half_type, shape::type_t::int8_type},
+                   i->get_shape().type());
+           }))
+            return;
+
+        static size_t counter = 0;
+        module_ref mm         = mpm.create_module("mlir_" + std::to_string(counter++));
+        mm->set_bypass();
+        auto [anchor_op, top_inputs] = fuse_input_ops_and_gemm_based_op(mm, conv_based_op);
+        mm->add_return({anchor_op});
+        mpm.get_module().replace_instruction(
+            conv_based_op, mlir_op{conv_based_op->get_operator()}, top_inputs, {mm});
+    }
+};
+
+MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_MLIR_ENABLE_OPS);
+bool is_self_decide() { return env(MIGRAPHX_MLIR_ENABLE_OPS::value()).empty(); }
+
+bool is_requested(std::string_view option)
+{
+    assert(enabled(MIGRAPHX_MLIR_ENABLE_OPS{}));
+    auto string_value = string_value_of(MIGRAPHX_MLIR_ENABLE_OPS{}, "");
+    static const char delim{','};
+    string_value.push_back(delim);
```

Comment:
Why is the delimiter added to the end?

Reply:
Good question. It is needed to handle a corner case - i.e., when a list of strings separated by `,` has only one value and, thus, there is no `,` at all. 

Maybe it is related to

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/ac96370b92b940fc2102b8b2aacf06ae1576a80a/src/include/migraphx/stringutils.hpp#L89

which just adds a white space to the end. I think it is supposed be `delim`. Probably, in all tests inly whitespaces are used as delimiters.

Reply:
Updated according the recent changes in `stringutils`. Done